zerver/lib/user_topics.py

```diff
@@ -147,26 +150,20 @@ def set_user_topic_visibility_policy_in_database(
     )
 
     if created:
-        return
+        return database_changed
 
     duplicate_request: bool = row.visibility_policy == visibility_policy
 
-    if duplicate_request and ignore_duplicate:
-        return
+    if duplicate_request:
+        logging.info("Request to set the visibility_policy to its current value.")
```

Comment:
This log line needs some details to be useful. I think `User {user_id} tried to set visibility_policy to its current value of {visibility_policy}` -- then we'll at least be able to observe trends in which users and/or which values are seeing this happen to motivate an investigation.

Reply:
I added a similar log for `UserTopic.VisibilityPolicy.INHERIT` case:
```python
    logging.info(
        "User %s tried to remove visibility_policy, which actually doesn't exist",
        user_profile.id,
    )
 ```